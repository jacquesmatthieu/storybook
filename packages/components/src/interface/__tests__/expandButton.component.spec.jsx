import React from 'react';
import { shallow } from 'enzyme';

import ExpandButton from '../expandButton.component';

describe('ExpandButton', () => {
  let props;

  beforeEach(() => {
    props = {};
  });

  const getWrapper = () => shallow(<ExpandButton {...props}>Foo</ExpandButton>);

  it('should render with default props', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with specified props', () => {
    // Given
    props.expanded = true;
    props.additionalProp = 'foo';

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
