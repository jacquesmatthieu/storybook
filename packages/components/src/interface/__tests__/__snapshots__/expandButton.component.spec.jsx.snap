// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandButton should render with default props 1`] = `
<expandButtoncomponent__ButtonOverride
  expanded={false}
  hollow={true}
>
  <expandButtoncomponent__ButtonInner>
    Foo
    <expandButtoncomponent__ExpandIcon
      expanded={false}
    >
      <Icon
        direction="down"
        name="arrow"
        theme={
          Object {
            "palette": Object {},
          }
        }
      />
    </expandButtoncomponent__ExpandIcon>
  </expandButtoncomponent__ButtonInner>
</expandButtoncomponent__ButtonOverride>
`;

exports[`ExpandButton should render with specified props 1`] = `
<expandButtoncomponent__ButtonOverride
  additionalProp="foo"
  expanded={true}
  hollow={true}
>
  <expandButtoncomponent__ButtonInner>
    Foo
    <expandButtoncomponent__ExpandIcon
      expanded={true}
    >
      <Icon
        direction="up"
        name="arrow"
        theme={
          Object {
            "palette": Object {},
          }
        }
      />
    </expandButtoncomponent__ExpandIcon>
  </expandButtoncomponent__ButtonInner>
</expandButtoncomponent__ButtonOverride>
`;
